src/service/resource.js

```diff
@@ -675,14 +661,38 @@ export class Resource {
       }
     } else {
       // Element is in viewport
-      this.resolveRenderOutsideViewport_();
       return true;
     }
-    const result = distance < viewportBox.height * multipler / scrollPenalty;
-    if (result) {
+    return distance < viewportBox.height * multipler / scrollPenalty;
+  }
+
+  /**
+   * Whether this is allowed to render when not in viewport.
+   * @return {boolean}
+   */
+  renderOutsideViewport() {
+    // The exception is for owned resources, since they only attempt to
+    // render outside viewport when the owner has explicitly allowed it.
+    // TODO(jridgewell, #5803): Resources should be asking owner if it can
+    // prerender this resource, so that it can avoid expensive elements wayyy
+    // outside of viewport. For now, blindly trust that owner knows what it's
+    // doing.
+    if (this.hasOwner() ||
+        this.withinViewportMultiplier_(this.element.renderOutsideViewport())) {
```

Comment:
Please split this into two if statements, so that the comment applies. Yes, duplication, but clarity.

Reply:
Done